app/models/amusement_park.rb

```diff
@@ -1,3 +1,11 @@
 class AmusementPark < ApplicationRecord
   has_many :rides
+  has_many :mechanics, through: :rides
+
+  validates :name, presence: true
+  validates :admission_cost, presence: true
+
+  def ride_mechanics
+    Mechanic.joins(:amusement_parks).distinct.where("amusement_parks.id = #{self.id}")
```

Comment:
An even easier option since you have that additional association set up on line 3: `mechanics.distinct`